localstack/services/iam/provider.py

```diff
@@ -496,3 +496,23 @@ def inline_policy_unapply_policy(fn, self, backend):
         except Exception:
             # Actually role can be deleted before policy being deleted in cloudformation
             pass
+
+    @patch(IAMBackend.create_policy)
+    def clean_policy_document_from_no_values(
```

Comment:
In my opinion, this should be part of the cloudformation model, not of IAM itself, since this is a CDK/cloudformation specific value, which has no notion in IAM itself, right? I would prefer not to mix those things here (unless there is indication IAM itself does this?)

Reply:
Yes, I thought the same but I wasn't sure. Just found that there's already a cleaning method implemented in another IAM model.

Reply:
This (or maybe we need more similar fixes for other models as well) might also fix #4517

Reply:
I added the same fix for inline policies of a role (also extended the test). Thanks for mentioning it.